stl/inc/chrono

```diff
@@ -617,6 +618,98 @@ namespace chrono {
     };
 
     using high_resolution_clock = steady_clock;
+
+#ifdef _HAS_CXX20
+    // [time.duration.io]
+    class _Units_suffix_helper {
+    public:
+        template <class _CharT, class _Period>
+        _NODISCARD static constexpr const _CharT* _Get_literal() {
+            for (const auto& _Entry : _Suffixes) {
+                if (_Entry._Num == _Period::num && _Entry._Den == _Period::den) {
+                    return _STD get<const _CharT*>(_Entry._Name);
+                }
+            }
+            return nullptr;
+        }
+
+        template <class _CharT>
+        _NODISCARD static _CharT* _Get_general(_CharT* _Rnext, intmax_t _Num, intmax_t _Den) {
+            // Returns the head pointer of the string, built in reverse.
+            _STL_INTERNAL_CHECK(_Num > 0 && _Den > 0);
+            *--_Rnext = '\0';
+            *--_Rnext = 's';
+            *--_Rnext = ']';
+            if (_Den != 1) {
+                _Rnext    = _UIntegral_to_buff(_Rnext, static_cast<uintmax_t>(_Den));
+                *--_Rnext = '/';
+            }
+
+            _Rnext    = _UIntegral_to_buff(_Rnext, static_cast<uintmax_t>(_Num));
+            *--_Rnext = '[';
+            return _Rnext;
+        }
+
+    private:
+        struct _Suf_names {
+            const intmax_t _Num;
+            const intmax_t _Den;
+            const pair<const char*, const wchar_t*> _Name;
+        };
+
+#define _LITERAL_SUFFIX(_TYPE, _SUFFIX) \
+    { _TYPE::num, _TYPE::den, _STD make_pair(_SUFFIX, L##_SUFFIX) }

```

Comment:
We should be able to use Class Template Argument Deduction and avoid the helper function call:
```suggestion
    { _TYPE::num, _TYPE::den, pair{_SUFFIX, L##_SUFFIX} }
```